src/middleware/etag/digest.ts

```diff
@@ -0,0 +1,42 @@
+const mergeBuffers = (buffer1: ArrayBuffer | undefined, buffer2: Uint8Array): Uint8Array => {
+  if (!buffer1) {
+    return buffer2
+  }
+  const merged = new Uint8Array(buffer1.byteLength + buffer2.byteLength)
+  merged.set(new Uint8Array(buffer1), 0)
+  merged.set(buffer2, buffer1.byteLength)
+  return merged
+}
+
+export const generateDigest = async (
+  stream: ReadableStream<Uint8Array> | null
+): Promise<string | null> => {
+  if (!stream || !crypto || !crypto.subtle) {
+    return null
+  }
+
+  let result: ArrayBuffer | undefined = undefined
+
+  const reader = stream.getReader()
+  for (;;) {
+    const { value, done } = await reader.read()
+    if (done) {
+      break
+    }
+
+    result = await crypto.subtle.digest(
+      {
+        name: 'SHA-1',
+      },
+      mergeBuffers(result, value)
+    )
+  }
+
+  if (!result) {
+    return null
```

Comment:
@usualoma 

Do you think we should test this line? You can write the test code like this:

```ts
it('Should return etag header when the steam is empty', async () => {
  const app = new Hono()
  app.use('/etag/*', etag())
  app.get('/etag/abc', (c) => {
    const stream = new ReadableStream({
      start(controller) {
        controller.close()
      },
    })
    return c.body(stream)
  })
  const res = await app.request('http://localhost/etag/abc')
  expect(res.status).toBe(200)
  expect(res.headers.get('ETag')).toBeNull()
})
```

Reply:
Thank you!
I've been added d7083e0

Reply:
Good!